client/checkout/blocks/upe-deferred-intent-creation/payment-processor.js

```diff
@@ -71,7 +78,69 @@ const PaymentProcessor = ( {
 		? testingInstructions
 		: '';
 	const gatewayConfig = getPaymentMethods()[ upeMethods[ paymentMethodId ] ];
-	const billingData = useCustomerData().billingAddress;
+	const customerData = useCustomerData();
+	const billingData = customerData.billingAddress;
+
+	useEffect( () => {
+		if (
+			paymentMethodsConfig.link !== undefined &&
+			paymentMethodsConfig.card !== undefined
+		) {
+			enableStripeLinkPaymentMethod( {
+				api: api,
+				elements: elements,
+				emailId: 'email',
+				fill_field_method: ( address, nodeId, key ) => {
+					const setAddress =
+						BLOCKS_SHIPPING_ADDRESS_FIELDS[ key ] === nodeId
+							? customerData.setShippingAddress
+							: customerData.setBillingData ||
+							  customerData.setBillingAddress;
+					const customerAddress =
+						BLOCKS_SHIPPING_ADDRESS_FIELDS[ key ] === nodeId
+							? customerData.shippingAddress
+							: customerData.billingData ||
+							  customerData.billingAddress;
+
+					if ( 'line1' === key ) {
+						customerAddress.address_1 = address.address[ key ];
+					} else if ( 'line2' === key ) {
+						customerAddress.address_2 = address.address[ key ];
+					} else if ( 'postal_code' === key ) {
+						customerAddress.postcode = address.address[ key ];
+					} else {
+						customerAddress[ key ] = address.address[ key ];
+					}
+
+					setAddress( customerAddress );
+
+					if ( customerData.billingData ) {
+						customerData.billingData.email = getBlocksEmailValue();
+						customerData.setBillingData( customerData.billingData );
+					} else {
+						customerData.billingAddress.email = getBlocksEmailValue();
+						customerData.setBillingAddress(
+							customerData.billingAddress
+						);
+					}
+				},
+				show_button: blocksShowLinkButtonHandler,
+				complete_shipping: () => {
+					return (
+						null !== document.getElementById( 'shipping-address_1' )
+					);
+				},
+				shipping_fields: BLOCKS_SHIPPING_ADDRESS_FIELDS,
+				billing_fields: BLOCKS_BILLING_ADDRESS_FIELDS,
+				complete_billing: () => {
+					return (
+						null !== document.getElementById( 'billing-address_1' )
+					);
+				},
+			} );
```

Comment:
This implementation already used a lot of copied code between gateway implementations. I tried to reduce this as much as possible, but sadly React would not let me remove the `fill_field_method` to another function, because `customerData` manipulates state data, so must be defined within the `useEffect` block. Consequently, `enableStripeLinkPaymentMethod` must be declared like this in all its hideous glory. 

It made me sad, but once we eventually have a single gateway, there won't be any more duplicates and that makes me a little more glad. 🤷